src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -276,6 +278,12 @@ sharded_store::get_versions(const subject& sub, include_deleted inc_del) {
     co_return std::move(versions).value();
 }
 
+ss::future<bool>
+sharded_store::is_referenced(const subject& sub, schema_version ver) {
+    auto map = [&sub, ver](store& s) { return s.is_referenced(sub, ver); };
+    co_return co_await _store.map_reduce0(map, false, std::logical_or<>{});
```

Comment:
should we send a copy of `sub` to each core rather than using a cross-core reference? it's not about correctness in this case, but rather patterns for high-performance

Reply:
I was curious about this. If it's not mutated, there's no cache invalidation, so there's no benefit? Only the cost of the copy?

Reply:
yeh it's like you're pushing / broadcasting a message to N cores vs having N cores pulling from one memory location. what's faster? I dunno, and it is probably situation dependent. I usually just send a copy unless it's a really big chunk of data.

Reply:
In the absence of a benchmark, I think I'll defer this.

Reply:
it's not so much about wanting the performance to be better here, it's about defaulting to safe and performant patterns. we just shouldn't be sharing data across cores without good reason

Reply:
I think the concern is sharing _mutable_ data. I'm happy to change it.

Reply:
mutability is a concern for _correctness_. what I'm saying is that by defaulting to safe and performant patterns we don't have to have to think about any of that, only if it is an exception based almost entirely on the size of the data. assuming we are talking about some strings, I think that's preferable. but there is not reason to change it now.